raco/algebra.py

```diff
@@ -66,6 +70,8 @@ def scheme(self):
     def walk(self):
         """Return an iterator over the tree of operators."""
         yield self
+        if self.stop_recursion:
```

Comment:
Yep they need it as well. One concern about adding this `stop_recursion` is that I'm not sure if I have added it to every necessary place. Right now I'm only relying on tests.